assets/src/edit-story/components/library/panes/pageLayouts/pageLayoutsPane.js

```diff
@@ -60,65 +61,69 @@ function PageLayoutsPane(props) {
     actions: { getTemplates },
   } = useAPI();
   const [templates, setTemplates] = useState([]);
-  const [selectedTemplateId, setSelectedTemplateId] = useState(null);
+  const [selectedPageLayoutType, setSelectedPageLayoutType] = useState(null);
 
   const pageLayoutsParentRef = useRef();
 
+  // load and process templates
   useEffect(() => {
     getTemplates().then((result) => setTemplates(result));
-  }, [getTemplates]);
+  }, [getTemplates, setTemplates]);
 
   const pills = useMemo(
     () =>
-      templates.map((template) => ({
-        id: template.id,
-        label: template.title,
+      Object.entries(PAGE_LAYOUT_TYPES).map(([key, { name }]) => ({
+        id: key,
+        label: name,
       })),
-    [templates]
+    []
   );
 
-  const filteredPages = useMemo(() => {
-    if (selectedTemplateId) {
-      const template = templates.find(
-        (template) => template.id === selectedTemplateId
-      );
-      if (template) {
-        return template.pages;
-      }
-    }
-    return templates.reduce(
-      (pages, template) => [
-        ...pages,
-        ...template.pages.map((page, index) => ({
-          ...page,
-          title: sprintf(
-            /* translators: 1: template name. 2: page number. */
-            __(`%1$s Page %2$s`, 'web-stories'),
-            template.title,
-            index + 1
-          ),
-        })),
-      ],
-      []
-    );
-  }, [templates, selectedTemplateId]);
-
-  const handleSelectTemplate = useCallback((templateId) => {
-    setSelectedTemplateId(templateId);
+  const filteredPages = useMemo(
+    () =>
+      templates.reduce((pages, template) => {
+        const templatePages = template.pages.reduce((acc, page) => {
+          // skip unselected page layout types if not matching
+          if (
+            !page.pageLayoutType ||
+            (selectedPageLayoutType &&
+              page.pageLayoutType !== selectedPageLayoutType)
+          ) {
+            return acc;
+          }
+
+          // translation not required because page layout title is already translated
+          const pageLayoutName = PAGE_LAYOUT_TYPES[page.pageLayoutType].name;
+          return [
+            ...acc,
+            {
+              ...page,
+              title: `${template.title} ${pageLayoutName}`,
```

Comment:
Still needs translation for RTL support I think.

This way translators could flip things if needed.

```js
sprintf(
  /* translators: 1: template name. 2: page layout name. */
  _x('%1$s %2$s', 'page layout title', 'web-stories'),
  template.title,
  pageLayoutName
)
```

Reply:
I mentioned in comments I pulled this out because `pageLayoutName` is already translated in the `constants.js` file. I can un-translate that and have it translate the combined here, but that creates a lot more translations than just translating the template name separately from the layout type. Thoughts?

Reply:
You can keep `pageLayoutName` translated like you already did, that is fine. But the `'%1$s %2$s'`peace should be translatable as well (if order needs to be changed by translators).

So we need both, not either or

Reply:
Okay, I'll bring this translation back.